libs/langchain/langchain/llms/fireworks.py

```diff
@@ -1,377 +1,90 @@
-"""Wrapper around Fireworks APIs"""
-import json
-import logging
-from typing import (
-    Any,
-    Dict,
-    List,
-    Optional,
-    Set,
-    Tuple,
-    Union,
-)
-
-import requests
+import os
+from typing import Any, Dict, Iterator, List, Optional
 
 from langchain.callbacks.manager import (
-    AsyncCallbackManagerForLLMRun,
     CallbackManagerForLLMRun,
 )
-from langchain.llms.base import BaseLLM
-from langchain.pydantic_v1 import Field, root_validator
-from langchain.schema import Generation, LLMResult
-from langchain.utils import get_from_dict_or_env
-
-logger = logging.getLogger(__name__)
-
-
-class BaseFireworks(BaseLLM):
-    """Wrapper around Fireworks large language models."""
-
-    model_id: str = Field("accounts/fireworks/models/llama-v2-7b-chat", alias="model")
-    """Model name to use."""
-    temperature: float = 0.7
-    """What sampling temperature to use."""
-    max_tokens: int = 512
-    """The maximum number of tokens to generate in the completion.
-    -1 returns as many tokens as possible given the prompt and
-    the models maximal context size."""
-    top_p: float = 1
-    """Total probability mass of tokens to consider at each step."""
-    fireworks_api_key: Optional[str] = None
-    """Api key to use fireworks API"""
-    batch_size: int = 20
-    """Batch size to use when passing multiple documents to generate."""
-    request_timeout: Optional[Union[float, Tuple[float, float]]] = None
-    """Timeout for requests to Fireworks completion API. Default is 600 seconds."""
-    max_retries: int = 6
-    """Maximum number of retries to make when generating."""
-
-    @property
-    def lc_secrets(self) -> Dict[str, str]:
-        return {"fireworks_api_key": "FIREWORKS_API_KEY"}
+from langchain.llms.base import LLM
+from langchain.schema.language_model import LanguageModelInput
+from langchain.schema.output import GenerationChunk
+from langchain.schema.runnable.config import RunnableConfig
+import openai
+
+
+def _stream_response_to_generation_chunk(
+    stream_response: Dict[str, Any],
+) -> GenerationChunk:
+    """Convert a stream response to a generation chunk."""
+    return GenerationChunk(
+        text=stream_response["choices"][0]["text"],
+        generation_info=dict(
+            finish_reason=stream_response["choices"][0].get("finish_reason", None),
+            logprobs=stream_response["choices"][0].get("logprobs", None),
+        ),
+    )
+
+
+class Fireworks(LLM):
```

Comment:
Shouldn't we add completion with retrying?

Reply:
openai create api has retrying itself: https://github.com/openai/openai-python/blob/5d50e9e3b39540af782ca24e65c290343d86e1a9/openai/api_resources/completion.py#L23